### PR TITLE
Quiz functionality

### PR DESCRIPTION
**Pull request by RahullRawat**
- User can play quiz
- After selecting one option all other options will be disabled
- Initially next button will be disabled, it will enabled only after you have selected any option
- Result page with correct answers and score will be displayed at the end of quiz.

**Files to check**
- Home.js
- Questions.js
- Result.js
- Quiz context and reducer

Preview Link - https://spectacular-basbousa-0f8cc9.netlify.app/

`Some code is commented intentionally will work on that in next PR`

### CODE REVIEW

File: src/pages/Home/Home.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+		quizDispatch,
+	} = useQuiz();
+
+	const startQuizHandler = (category) => {
+		// if (token) {
+		navigate(`/rules/${category}`);
+		quizDispatch({ type: "RESET" });
+		// } else {
+		// navigate("/login");
+		// }
```
Comment (gadgetfather) [inline src/pages/Home/Home.js:24]: Remove commented code if not needed

File: src/pages/Questions/Questions.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
+import React, { useState } from "react";
+import { useParams, useNavigate } from "react-router-dom";
+import { useQuiz } from "../../context/QuizContext";
+import "./Questions.css";
+
+export const Questions = () => {
+	const [currentQuestion, setCurrentQuestion] = useState(0);
+	const [selected, setSelected] = useState();
+	const [optionDisabled, setOptionDisabled] = useState(false);
+	const [btnDisabled, setBtnDisabled] = useState(true);
```
Comment (gadgetfather) [inline src/pages/Questions/Questions.js:10]: You can use useReducer to handle multiple states

File: src/pages/Questions/Questions.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+							(ele, index) => {
+								return (
+									<button
+										key={index}
+										onClick={() => optionClickHandler(ele.option)}
+										className={`${selected === ele.option ? "selected" : ""}`}
+										disabled={optionDisabled}
+									>
+										{ele.option}
+									</button>
```
Comment (gadgetfather) [inline src/pages/Questions/Questions.js:70]: You can style this button differently when it is disabled. So the user can know that he/she cannot proceed without selecting an answer.

File: src/pages/Result/Result.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+									</div>
+								</div>
+							</div>
+						);
+					})}
+				</div>
+			</div>
+		</div>
+	);
+};
```
Comment (gadgetfather) [inline src/pages/Result/Result.js:55]: Format code neatly.

**Review by gadgetfather — APPROVED** [review]
- Good file structure
- Functionalities are also working as expected
- Nice UI for the result section
Apart from some suggestions, PR is good to merge. 
Great work!

**Review by AtulPant2704 — APPROVED** [review]

**Pull request merged**